controllers/managedocs_controller_test.go

```diff
@@ -474,1153 +474,1265 @@ var _ = Describe("ManagedOCS controller", func() {
-					return k8sClient.Get(ctx, key, managedOCS)
-				}, timeout, interval).Should(Succeed())
+					managedOCS := managedOCSTemplate.DeepCopy()
+					key := utils.GetResourceKey((managedOCS))
+					Consistently(func() error {
+						return k8sClient.Get(ctx, key, managedOCS)
+					}, timeout, interval).Should(Succeed())
+				})
 			})
-		})
-		When("there are pvcs in a secondary namespace while all other uninstall conditions are met", func() {
-			It("should not delete the managedOCS resource", func() {
-				setupUninstallConditions(true, testAddonConfigMapDeleteLabelKey, true, true, false, false, true)
+			When("there are pvcs in a secondary namespace while all other uninstall conditions are met", func() {
+				It("should not delete the managedOCS resource", func() {
+					setupUninstallConditions(true, testAddonConfigMapDeleteLabelKey, true, true, false, false, true)
 
-				managedOCS := managedOCSTemplate.DeepCopy()
-				key := utils.GetResourceKey((managedOCS))
-				Consistently(func() error {
-					return k8sClient.Get(ctx, key, managedOCS)
-				}, timeout, interval).Should(Succeed())
+					managedOCS := managedOCSTemplate.DeepCopy()
+					key := utils.GetResourceKey((managedOCS))
+					Consistently(func() error {
+						return k8sClient.Get(ctx, key, managedOCS)
+					}, timeout, interval).Should(Succeed())
+				})
 			})
-		})
-		When("All uninstall conditions are met", func() {
-			It("should delete the managedOCS", func() {
-				setupUninstallConditions(true, testAddonConfigMapDeleteLabelKey, true, true, true, false, false)
-
-				managedOCS := managedOCSTemplate.DeepCopy()
-				key := utils.GetResourceKey((managedOCS))
-				Eventually(func() bool {
-					err := k8sClient.Get(ctx, key, managedOCS)
-					return err != nil && errors.IsNotFound(err)
-				}, timeout, interval).Should(BeTrue())
-			})
-			It("should delete the deployer csv", func() {
-				csv := csvTemplate.DeepCopy()
-				key := utils.GetResourceKey(csv)
-				Eventually(func() bool {
-					err := k8sClient.Get(ctx, key, csv)
-					return err != nil && errors.IsNotFound(err)
-				}, timeout, interval).Should(BeTrue())
+			When("All uninstall conditions are met", func() {
+				It("should delete the managedOCS", func() {
+					setupUninstallConditions(true, testAddonConfigMapDeleteLabelKey, true, true, true, false, false)
+
+					managedOCS := managedOCSTemplate.DeepCopy()
+					key := utils.GetResourceKey((managedOCS))
+					Eventually(func() bool {
+						err := k8sClient.Get(ctx, key, managedOCS)
+						return err != nil && errors.IsNotFound(err)
+					}, timeout, interval).Should(BeTrue())
+				})
+				It("should delete the deployer csv", func() {
+					csv := csvTemplate.DeepCopy()
+					key := utils.GetResourceKey(csv)
+					Eventually(func() bool {
+						err := k8sClient.Get(ctx, key, csv)
+						return err != nil && errors.IsNotFound(err)
+					}, timeout, interval).Should(BeTrue())
+				})
+			})
+			When("the uninstall condition is met", func() {
```

Comment:
- nit: this doesn't read correctly since we are removing the left over resources manually
- rephrasing to state operation is being done manually would be good, only if possible.